WooCommerce/Classes/Blaze/BlazeLocalNotificationScheduler.swift

```diff
@@ -0,0 +1,166 @@
+            return
+        }
+
+        observeStorageAndScheduleNotifications()
+
+        pushNotesManager.localNotificationUserResponses
+            .sink { [weak self] response in
+                guard let self,
+                      response.notification.request.identifier == LocalNotification.Scenario.blazeNoCampaignReminder.identifier else {
+                    return
+                }
+
+                userDefaults.setBlazeNoCampaignReminderOpened(true)
+            }
+            .store(in: &subscriptions)
+    }
+}
+
+private extension DefaultBlazeLocalNotificationScheduler {
+    func isEligibleForBlaze() async -> Bool {
+        guard let site = stores.sessionManager.defaultSite else {
+            return false
+        }
+        return await blazeEligibilityChecker.isSiteEligible(site)
+    }
+
+    /// Performs initial fetch from storage and updates results.
+    func observeStorageAndScheduleNotifications() {
+        blazeCampaignResultsController.onDidChangeContent = { [weak self] in
+            self?.scheduleLocalNotificationIfNeeded()
+        }
+        blazeCampaignResultsController.onDidResetContent = { [weak self] in
+            self?.scheduleLocalNotificationIfNeeded()
+        }
+
+        do {
+            try blazeCampaignResultsController.performFetch()
+            scheduleLocalNotificationIfNeeded()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
+    }
+
+    func scheduleLocalNotificationIfNeeded() {
+        guard !userDefaults.blazeNoCampaignReminderOpened() else {
+            DDLogDebug("Blaze: User interacted with a previously scheduled no campaign local notification. Don't schedule again.")
+            return
+        }
+
+        let campaigns = blazeCampaignResultsController.fetchedObjects
+        guard campaigns.contains(where: { $0.isEvergreen && $0.isActive}) == false else {
+            Task { @MainActor in
+                await scheduler.cancel(scenario: .blazeNoCampaignReminder)
+            }
+            return DDLogDebug("Blaze: An active evergreen campaign is present. No need to schedule a local notification.")
+        }
+
+        let latestEndTime = campaigns
+            .filter({ $0.isEvergreen == false })
+            .map({ campaign -> Date? in
+                guard let startTime = campaign.startTime else {
+                    return nil
+                }
+                let durationDays = Int(campaign.durationDays)
+                return Calendar.current.date(byAdding: .day, value: durationDays, to: startTime)
+            })
+            .compactMap { $0 }
+            .max()
+
+        guard let latestEndTime,
+              let notificationTime = Calendar.current.date(byAdding: .day, value: Constants.daysDurationNoCampaignReminderNotification, to: latestEndTime) else {
+            return DDLogDebug("Blaze: Failed calculating notification time from latest campaign end time.")
+        }
```

Comment:
Regarding my previous nit-picking, I meant declaring `notificationTime` separately as it's long and hard to read in a guard check:

```suggestion
        let notificationTime = Calendar.current.date(byAdding: .day, value: Constants.daysDurationNoCampaignReminderNotification, to: latestEndTime)
        guard let latestEndTime, let notificationTime else {
            return DDLogDebug("Blaze: Failed calculating notification time from latest campaign end time.")
        }
```

Reply:
Updated in https://github.com/woocommerce/woocommerce-ios/pull/13816/commits/ad51983dbf6ca7182add02d296054a1289d92978